Documentation/auto-dependency-flow/api.md

```diff
@@ -141,22 +141,20 @@ Use the AssetRoot properties if they are defined. Example (adapted from [CLI's b
 
   <PropertyGroup>
     <CoreSetupBlobRootUrl Condition="'$(DotNetAssetRootUrl)' != ''">$(DotNetAssetRootUrl)</CoreSetupBlobRootUrl>
-    <CoreSetupBlobRootUrl Condition="'$(CoreSetupBlobRootUrl)' == ''">https://dotnetcli.azureedge.net/dotnet/</CoreSetupBlobRootUrl>
-
-    <CoreSetupRootUrl>$(CoreSetupBlobRootUrl)Runtime/</CoreSetupRootUrl>
+    <CoreSetupBlobRootUrl Condition="'$(CoreSetupBlobRootUrl)' == ''">https://dotnetcli.azureedge.net/dotnet/Runtime/$(SharedFrameworkVersion)/</CoreSetupBlobRootUrl>
```

Comment:
Why are we hard-coding this here? I would expect it to be in the build definition.

Reply:
a) that's how the CLI works today.  https://github.com/dotnet/cli/blob/86732427fa598a72381a103dda83585e9938867b/build/BundledRuntimes.props#L27
b) This is just an example in docs.  The real code can do whatever it wants.

Reply:
lol - I missed this is just a doc but still my question holds that I would like to avoid hard-coding these paths here as I don't like the duplication of this data in the build definitions and in code. We should just pass it from the build definition, particularly when we need to override it for our servicing builds.

Reply:
IMO it is easier to understand what we are trying to illustrate here - how to use `$(DotNetAssetRootUrl)` - if we limit the number of concepts we are using. Trying to describe some other external "build definition parameter", which serves no purpose here just clutters the example.

Reply:
Also, this URL is necessary for the dev machine scenario when the developer just types "build" and doesn't supply the URL.

Reply:
I'm not sure if you agree or disagree with my suggestion :)

At any rate my primary concern is that we avoid duplication of this information in various places and I don't think it is actually needed for local dev builds so I would prefer they stay as a build definition parameter.

Reply:
CLI's in a state where I think it's actually more maintainable if the default stays in the source code: CLI has no checked-in build defs and no central build definition to set the variable in. Moving to the build definitions would actually duplicate this URL across each platform leg.

Reply:
I don't think you are understanding what is happening here.

The CLI needs to get the Shared Framework from Core-Setup during its build.  When a developer just has the CLI repo and types "build", the CLI build needs to acquire the Shared Framework from somewhere.

> I don't think it is actually needed for local dev builds

This is where you are misunderstanding.  It is 100% needed.  You can't build the CLI repo without a runtime.

Reply:
Think of this like myget URLs.  We need to get the NuGet packages from somewhere.  You can't build without them.  By default, we hard-code the URLs to the feeds we are acquiring the NuGet packages from.  This is the exact same thing.

Reply:
Yes I am miss-understanding :) I was thinking that this is where we are publishing stuff from core-setup, however now if I understand this is essentially like a checked in feed for folks like CLI to be able to pull the shared framework. I think that is OK. 

Hangs head in sham and will go read the docs closer :(